src/v2/Carousel/Carousel.sol

```diff
@@ -251,11 +251,11 @@ contract Carousel is VaultV2 {
         // check if user has enough balance
         if (balanceOf(_receiver, _epochId) < _assets)
             revert InsufficientBalance();
-
+        
         // check if user has already queued up a rollover
-        if (ownerToRollOverQueueIndex[_receiver] != 0) {
-            // if so, update the queue
+        if (isEnlistedInRolloverQueue(_receiver)) {
```

Comment:
Using this will create a new QueueItem for delisted user instead of repopulating (since isEnlistedInRolloverQueue will return false if assets == 0), making it impossible to ever clean them up after